mypy/typeanal.py

```diff
@@ -239,10 +239,9 @@ def visit_unbound_type(self, t: UnboundType) -> Type:
                     self.fail('Invalid type "{}"'.format(name), t)
                 return t
             info = sym.node  # type: TypeInfo
-            if len(t.args) > 0 and info.fullname() == 'builtins.tuple':
-                return TupleType(self.anal_array(t.args),
-                                 Instance(info, [AnyType()], t.line),
-                                 t.line)
+            if info.fullname() == 'builtins.tuple':
+                assert not t.args
+                return Instance(info, self.anal_array([AnyType()]), t.line, t.column)
```

Comment:
This change looks a bit strange. Could you please add a comment explaining this?